contrib/kafka/Readme.md

```diff
@@ -0,0 +1,103 @@
+# Knative KafkaEventSource
+
+This deploys an Operator that can instantiate Knative Event Sources which subscribe to a Kafka Topic.
+
+## EventSource
+
+The image that receives messages from Kafka (Receive Adaptor) is available at `docker.io/sjwoodman/kafkaeventsource`.
+To build it from source edit the `eventsource/Makefile` with suitable tags for your DockerHub organisation and perform the following commands.
+
+```bash
+make docker_build docker_push_latest
+```
+
+You will need to edit the _Operator_ with the location of the newly built image.
+
+```bash
+sed -i 's|sjwoodman|DH_ORG|g' kafkaeventsource-operator/deploy/operator.yaml
+sed -i 's|sjwoodman|DH_ORG|g' kafkaeventsource-operator/pkg/controller/kafkaeventsource
+```
+
+## Usage
+
+The sample connects to a [Strimzi](http://strimzi.io/quickstarts/okd/) Kafka Broker running inside OpenShift.
+
+1. Setup [Knative Eventing](https://github.com/knative/docs/tree/master/eventing).
+
+1. Install the [in-memory `ClusterChannelProvisioner`](https://github.com/knative/eventing/tree/master/config/provisioners/in-memory-channel).
+    - Note that you can skip this if you choose to use a different type of `Channel`. If so, you will need to modify `01-channel.yaml` before deploying it.
+
+1. Install the KafkaEventSource CRD
+
+    ```bash
+    kubectl create -f deploy/crds/sources_v1alpha1_kafkaeventsource_crd.yaml
+    ```
+
+1. Setup RBAC and deploy the KafkaEventSource-operator:
+
+    ```bash
+    kubectl create -f deploy/service_account.yaml
+    kubectl create -f deploy/role.yaml
+    kubectl create -f deploy/role_binding.yaml
+    kubectl create -f deploy/operator.yaml
+    ```
+
+1. Verify that the KafkaEventSource-operator is up and running:
+
+    ```bash
+    $ kubectl get deployment
+    NAME                              DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+    kafkaeventsource-operator         1         1         1            1           2m
+    ```
+
+1. (Optional) Create a Topic in Strimzi to back the Channel.
+    If you don't do this, defaults will be used for the number of partitions etc.
+
+    ```bash
+    kubectl create -f sample/00-topic.yaml
+    ```
+
+1. Create a KafkaEventSource and wire it up to a function. Either
```

Comment:
Should state that the below scripts can be used to create a channel, service, or subscription. However, the event source is needed in any case.